file_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"path/filepath"
+	"io/ioutil"
+	"fmt"
+)
+
+// Although other tests besides those in this file require this env var, this init() func will cover all tests.
+func init() {
+	if os.Getenv("GITHUB_OAUTH_TOKEN") == "" {
+		fmt.Println("ERROR: These tests require that env var GITHUB_OAUTH_TOKEN be set to a GitHub Personal Access Token.")
+		fmt.Println("See the tests cases to see which GitHub repos the oAuth token needs access to.")
+		os.Exit(1)
```

Comment:
When you call `os.Exit(1)`, do any functions that were called with `defer` have a chance to run? I ask because, with `t.Parallel()`, it's possible that some of the test cases will have started running and possibly creating random files on the disk, when this `init` method kills the whole test without running any cleanup.